recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+import os
+from os.path import join
+from glob import glob
+from itertools import chain
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class PclConanRecipe(ConanFile):
+    name = "pcl"
+    description = "Point Cloud Library"
+    license = "BSD-3-Clause"
+    homepage = "https://pointclouds.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cuda": [True, False],
+        "with_tools": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "with_cuda": False,
+        "with_tools": False
+    }
+    requires = ("boost/1.70.0",
+                "eigen/3.3.7",
+                "flann/1.9.1",
+                "libpng/1.6.37",
+                "opengl/system",
+                "qhull/7.3.2")
+    generators = ["cmake", "cmake_find_package"]
+    exports = ["CMakeLists.txt"]
+
+    @property
+    def _source_subfolder(self):
+        return "pcl-pcl-{}".format(self.version)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if tools.msvs_toolset(self) == "v140":
+            raise ConanInvalidConfiguration("Unsupported Visual Studio Compiler or Toolset")
+        tools.check_min_cppstd(self, "14")
```

Comment:
```suggestion
        minimal_cpp_standard = "14"
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, minimal_cpp_standard)
        minimal_version = {
            "gcc": "5",
            "clang": "3.4",
            "apple-clang": "10",
            "Visual Studio": "14"
        }
        compiler = str(self.settings.compiler)
        if compiler not in minimal_version:
            self.output.warn(
                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
            self.output.warn(
                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
            return
        version = tools.Version(self.settings.compiler.version)
        if version < minimal_version[compiler]:
            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))

```
Run check_min_cppstd ONLY if cppstd is defined, which is not true by default.